src/Platform/Platform.cpp

```diff
@@ -2502,7 +2502,7 @@ GCodeResult Platform::ReceiveI2cOrModbus(GCodeBuffer& gb, const StringRef &reply
 						break;
 					}
 				}
-				else
+				else if(!gb.Seen('V')) // Report comm error if output variable not set
```

Comment:
@dc42 Thanks, good point on the condition style.

On line 2512, my understanding there was that it would only happen if a bad parameter were given or the lock on the UART could not be acquired - which _should_ (?) not occur under normal circumstances, so it might still be useful output when writing to variable.

Given that we can still check for any failure by looking at the variable value, I've changed both outputs to be conditional.